docs/compute_resources.md

```diff
@@ -0,0 +1,173 @@
+each resource type (CPU and memory), the sum of the resource limits of the containers scheduled to
+the node is less than the capacity of the node.  Note that although actual memory or CPU resource
+usage on nodes is very low, the scheduler will still refuse to place pods onto nodes if the capacity
+check fails.  This protects against a resource shortage on a node when resource usage later increases,
+such as due to a daily peak in request rate.
+
+If a pod does not specify a resource limit, then the scheduler treats the limit as zero.
+The scheduler also tries to spread pods evenly across nodes, among other scheduling
+considerations.  As a result, even if pods do not specify resource limits, the system may
+still function fine.  
+
+## How Pods with Resource Limits are Run
+
+When kubelet starts a container of a pod, it passes the CPU and memory limits to the container
+runner (Docker or rkt).
+
+When using Docker:
+- The `spec.container[].resources.limits.cpu` is multiplied by 1024, converted to an integer, and
+  used as the value of the [`--cpu-shares`](
+  https://docs.docker.com/reference/run/#runtime-constraints-on-resources) flag to the `docker run`
+  command.
+- The `spec.container[].resources.limits.memory` is converted to an integer, and used as the value
+  of the [`--memory`](https://docs.docker.com/reference/run/#runtime-constraints-on-resources) flag
+  to the `docker run` command.
+
+**TODO: document behavior for rkt**
+
+If a container exceeds its memory limit, it may be terminated.  If it is restartable, it will be
+restarted by kubelet, as will any other type of runtime failure.  If it is killed for exceeding its
+memory limit, you will see the reason `OOM Killed`, as in this example:
+```
+$ kubectl get pods/memhog
+NAME      READY     REASON       RESTARTS   AGE
+memhog    0/1       OOM Killed   0          1h
+```
+
+A container may or may not be allowed to exceed its CPU limit for extended periods of time.
+However, it will not be killed for excessive CPU usage.
+
+## Specifying Resource Quantities
+A resource quantity (an amount of CPU or memory) can be specified as an integer,
+a floating point value, or a string.  As a string, it can optionally use an SI suffix
+(E, P, T, G, M, k, m) or a their power-of-two analog (Ei, Pi, Ti, Gi, Mi, Ki).
+For example, the following JSON snippets are all valid and represent roughly the same value:
+```json
+   "cpu": 128974848,
+   "cpu": "128974848",
+   "cpu": 128974848.1,
+   "cpu": "128974848.1",
+   "cpu": 129e6,
+   "cpu": "129e6",
+   "cpu": "129M",
+   "cpu": "123Mi"
+```
+Quantities are converted to a fixed-precision string representation when pods are created or
+updated., and may be truncated to 3 decimal places in the process.
+
+## Monitoring Compute Resource Usage
+
+The resource usage of a pod is intentionally not reported when your get the pod
+resource, as this would cause the resource to change too quickly.
```

Comment:
s/as this would cause the resource to change too quickly/as the usage data could change quickly and presents too much write load for the pod resource/

?